File: Inflearn/5_완전탐색(DFS)/재귀함수를 이용한 이진수 출력.py
import sys


def DFS(x):
    if x >= 1:
        DFS(x//2)
        answer.append(x % 2)


if __name__ == "__main__":
    n = int(sys.stdin.readline())
    answer = []
    DFS(n)
    for x in answer:
        print(x, end='')

#ref
sys.stdin = open("input.txt", "r")


def DFS(x):
    if x == 0:
        return
    else:
        DFS(x//2)
        print(x % 2, end='')


if __name__ == "__main__":
    n = int(input())
    DFS(n)

File: JavaScript/Inflearn/8_재귀함수와 완전탐색_DFS/재귀함수를 이용한 이진수 출력.js
function solution(n) {
  let answer = '';

  function DFS(n) {
    if (n === 0) return;
    else {
      DFS(parseInt(n / 2));
      answer += n % 2; // String()으로 변환해도 됨
    }
  }

  DFS(n);

  return answer;
}

console.log(solution(11)); // 1011
